src/Engine/Data/RenderParameters.cpp

```diff
@@ -87,6 +112,20 @@ void RenderParameters::addParameter( const std::string& name, Data::Texture* tex
     m_texParamsVector[name] = TextureParameter( name, tex, texUnit );
 }
 
+void RenderParameters::addParameter( const std::string& name, const std::string& value ) {
+    auto it = m_enumConverters.find( name );
+    if ( it != m_enumConverters.end() ) { it->second->setEnumValue( *this, name, value ); }
```

Comment:
```suggestion
   addParameter( name, it->second->getEnumerator( value );
```
Not tested, but seems clearer for me (I don't get why the converter should set the value, instead of the parameter set get the conversion results and set it).

Reply:
Asking the converter to set the value (instead of setting it directly) allows to preserve the underlying type of the enumeration. 

See the documentation about getEnumerator : 
```
* \note  This method does not respect the underlying type of enumerations as it returns
* always an int. Use setEnumValue to modify the parameter set in a type safe way.
```